src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -271,8 +288,14 @@ void QgsRelationReferenceWidget::setForeignKey( const QVariant &value )
   {
     // Attributes from the referencing layer
     QgsAttributes attrs = QgsAttributes( mReferencingLayer->fields().count() );
-    // Set the value on the foreign key field of the referencing record
-    attrs[ mReferencingLayer->fields().lookupField( mRelation.fieldPairs().at( 0 ).first )] = value;
+    // Set the value on the foreign key fields of the referencing record
+
+    const QList<QgsRelation::FieldPair> fieldPairs = mRelation.fieldPairs();
+    for ( int i = 0; i < fieldPairs.count(); i++ )
```

Comment:
Better save than sorry

```suggestion
    int fieldCount = std::min( fieldPairs.count(), values.count() );
    for ( int i = 0; i < fieldCount; i++ )
```